src/raft.rs

```diff
@@ -1269,7 +1301,7 @@ impl<T: Storage> Raft<T> {
                 to_send.set_msg_type(MessageType::MsgReadIndexResp);
                 to_send.set_index(rs.index);
                 to_send.set_entries(req.take_entries());
-                *more_to_send = Some(to_send);
+                ctx.more_to_send.push(to_send);
```

Comment:
How about `more_to_send: HashMap<u64, Message>`? So we can send only 1 read index response to one peer.

Reply:
`HashMap` is slower than `Vec`, it may not be a good idea. On the other hand, the optimization assumes there is a queue in follower side, which may not be a valid case.